src/test/java/seedu/address/logic/parser/BuyCommandParserTest.java

```diff
@@ -0,0 +1,98 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.commands.CommandTestUtil.INVALID_GOODS_EMPTY;
+import static seedu.address.logic.commands.CommandTestUtil.INVALID_PRICE;
+import static seedu.address.logic.commands.CommandTestUtil.INVALID_QUANTITY;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_GOODS;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_GOODS_BUY_ORANGE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_GOODS_SELL_PAPAYA;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_PRICE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_PRICE_BUY_ORANGE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_PRICE_SELL_PAPAYA;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_QUANTITY;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_QUANTITY_BUY_ORANGE;
+import static seedu.address.logic.commands.CommandTestUtil.VALID_QUANTITY_SELL_PAPAYA;
+import static seedu.address.logic.parser.CommandParserTestUtil.assertParseFailure;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.address.logic.commands.BuyCommand;
+import seedu.address.model.transaction.Goods;
+
+
+public class BuyCommandParserTest {
+
+    private static final String MESSAGE_INVALID_FORMAT =
+            String.format(MESSAGE_INVALID_COMMAND_FORMAT, BuyCommand.MESSAGE_USAGE);
+    private BuyCommandParser parser = new BuyCommandParser();
+
+    @Test
+    public void parse_invalidIndexPreamble_failure() {
+        // negative index
+        assertParseFailure(parser, "-5", MESSAGE_INVALID_FORMAT);
+
+        // zero index
+        assertParseFailure(parser, "0", MESSAGE_INVALID_FORMAT);
+
+
+        // invalid arguments being parsed as preamble
+        assertParseFailure(parser, "1 some random string", MESSAGE_INVALID_FORMAT);
+
+        // invalid prefix being parsed as preamble
+        assertParseFailure(parser, "1 i/ string", MESSAGE_INVALID_FORMAT);
+    }
+
+
+    //DONE
```

Comment:
Would this comment be useful? Or do you think it can be removed 🤔

Reply:
Thanks for spotting this ! Will make the necessary changes.